[Telink] Extend CI (Zephyr 3.3 & 3.7 support) & Update compatible builds to docker version 125

s07641069 opened the pull request:
**Change overview**

- Extend CI (Zephyr 3.3 & 3.7 support)
- Update Readme files
- Update compatible builds to docker version 125 (from 119):
    - 120: [NXP] Update NXP docker image: https://github.com/project-chip/connectedhomeip/pull/37822
    - 121: [Telink] Update Docker image (Zephyr 3.3 & 3.7 support): https://github.com/project-chip/connectedhomeip/pull/37852
    - 122: [Telink] Update Docker image (Zephyr update): https://github.com/project-chip/connectedhomeip/pull/37982
    - 123: [Tizen] Add arm64 target - Docker part: https://github.com/project-chip/connectedhomeip/pull/37761
    - 124: [NXP] Update NXP docker image: https://github.com/project-chip/connectedhomeip/pull/38055
    - 125: [Tizen] Fix SDK installer when more than one CPU is selected: https://github.com/project-chip/connectedhomeip/pull/38098
    - Keep [TI CC32XX] image on version 115

| Name (Zephyr 3.3/Zephyr 3.7) | Target (Zephyr 3.3/Zephyr 3.7) | CI Zephyr 3.3 | CI Zephyr 3.7 |
|------------------------------|--------------------------------|---------------|---------------|
| Build example Telink (B92 retention) Air Quality Sensor App | telink-tlsr9528a_retention-air-quality-sensor | Run test for master and s07641069 PRs | Run test for master and s07641069 PRs |
| Build example Telink (W91) All Clusters App | telink-tlsr9118bdk40d-all-clusters | Run test for master and s07641069 PRs | Run test for master and s07641069 PRs |
| Build example Telink (B92) All Clusters Minimal App | telink-tlsr9528a-all-clusters-minimal | Run test for master and s07641069 PRs | Run test for master and s07641069 PRs |
| Build example Telink (tl721x) Bridge App | telink-tl7218x-bridge | Run test for master and s07641069 PRs | Run test for master and s07641069 PRs |
| Build example Telink (B92 retention) Contact Sensor App | telink-tlsr9528a_retention-contact-sensor | Run test for master and s07641069 PRs | Run test for master and s07641069 PRs |
| Build example Telink (W91) Lighting App with OTA, Factory Data | telink-tlsr9118bdk40d-light-ota-factory-data | Run test for master and all PRs | Run test for master and all PRs |
| Build example Telink (B91) Lighting App with OTA, RPC, Factory Data, 4Mb flash | telink-tlsr9518adk80d-light-ota-rpc-factory-data-4mb | Run test for master and all PRs | Run test for master and all PRs |
| Build example Telink (tl321x) Lighting App with OTA (LZMA), Shell, Factory Data | telink-tl3218x-light-ota-compress-lzma-shell-factory-data | Run test for master and all PRs | Run test for master and all PRs |
| Build example Telink (tl321x_retention) Light Switch App with OTA, Shell, Factory Data | telink-tl3218x_retention-light-switch-ota-shell-factory-data | Run test for master and all PRs | NO |
| Build example Telink (tl721x) Lighting App with OTA, Shell, Factory Data | telink-tl7218x-light-ota-shell-factory-data | Run test for master and all PRs | Run test for master and all PRs |
| Build example Telink (tl721x retention) Light Switch App with OTA (LZMA), Factory Data | telink-tl7218x_retention-light-switch-ota-compress-lzma-factory-data | Run test for master and all PRs | Run test for master and all PRs |
| Build example Telink (B92) Light Switch App with OTA (LZMA), Shell, Factory Data | telink-tlsr9528a-light-switch-ota-compress-lzma-shell-factory-data | Run test for master and all PRs | Run test for master and all PRs |
| Build example Telink (B92) Lock App with DFU | telink-tlsr9528a-lock-dfu | Run test for master and s07641069 PRs | Run test for master and s07641069 PRs |
| Build example Telink (tl321x) OTA Requestor App | telink-tl3218x-ota-requestor | Run test for master and s07641069 PRs | Run test for master and s07641069 PRs |
| Build example Telink (B91 USB) Pump App | telink-tlsr9518adk80d-pump-usb | Run test for master and s07641069 PRs | Run test for master and s07641069 PRs |
| Build example Telink (B91) Pump Controller App | telink-tlsr9518adk80d-pump-controller | Run test for master and s07641069 PRs | Run test for master and s07641069 PRs |
| Build example Telink (B91) Shell App | telink-tlsr9518adk80d-shell | Run test for master and s07641069 PRs | Run test for master and s07641069 PRs |
| Build example Telink (B92 retention) Smoke CO Alarm App | telink-tlsr9528a_retention-smoke-co-alarm | Run test for master and s07641069 PRs | Run test for master and s07641069 PRs |
| Build example Telink (B91 Mars) Temperature Measurement App with OTA | telink-tlsr9518adk80d-temperature-measurement-mars-ota | Run test for master and s07641069 PRs | Run test for master and s07641069 PRs |
| Build example Telink (B91) Thermostat App | telink-tlsr9518adk80d-thermostat | Run test for master and s07641069 PRs | Run test for master and s07641069 PRs |
| Build example Telink (tl721x) Window Covering App | telink-tl7218x-window-covering | Run test for master and s07641069 PRs | Run test for master and s07641069 PRs |

#### Testing
Verified by GitHub CI

s07641069 committed:
[Telink] Manual Zephyr SDK update to check CI jobs

s07641069 committed:
[Telink] Fix build errors:
- STREAM_FLASH_ERASE has direct dependencies STREAM_FLASH
- GETOPT_LONG multiple definition error

s07641069 committed:
[Telink] CONFIG_BOOT_MAX_IMG_SECTORS_AUTO used

s07641069 committed:
[Telink] Fix build errors

s07641069 committed:
[Telink] Reduce NET_BUF_*_COUNT (to be tested)

interfer committed:
[Telink] Zephyr v3.7 update changes

Adopted changes for Telink B9x board series for
upcoming update of Zephyr version.

Fixed err output from InitChipStack in mainCommon.
Tested advertisement tested manually.

s07641069 committed:
[Telink] Reduce mcuboot partition size to 60Kb
- Apply change for flash overlays <=2Mb
- Possible after reducing CONFIG_BOOT_MAX_IMG_SECTORS

s07641069 committed:
[Telink] avoid using NEWLIB_LIBC (use PICOLIB)

restyled-commits committed:
Restyled by whitespace

restyled-commits committed:
Restyled by clang-format

s07641069 committed:
[Telink] Update Zephyr revision & SDK

s07641069 committed:
[Telink] Adopt TL3218 to latest Zephyr version

s07641069 committed:
[Telink] Use 4mb flash for (B92) Light Switch App target build

s07641069 committed:
[Telink] Fix W91 target build

s07641069 committed:
[Telink] Fix B91 Mars target build

s07641069 committed:
[Telink] Use correct if SOC_RISCV_TELINK_TL321X

s07641069 committed:
[Telink] Fix retention target builds

interfer committed:
[Telink] fixed W91 WiFi interface init

reverted GetWifiInterface in InetUtils changes,
added syntax fixes to Kconfigs.default

s07641069 committed:
[Telink] Update to latest zephyr (develop_new_matter_fixes) hash

interfer committed:
[Telink] memory layout and default config changes

Reduced mcuboot partition by 0x1000 and removed two other partitions
getting 0x3000 to distribute 0x4000 between slot0 and slot 1.

Kconfig.defaults changes to reduce ROM usage on B92 lighting OTA.
To be improved

[Telink] reduced MCUboot partition for most targets

increase of slot0,1, bootloader config changes

[Telink] Fixed B9x platform startup

Lack of MBEDTLS_ZEPHYR_ENTROPY config in defaults
which led to InitChipStack 172 error.
As result entropy failed in src/platform/Zephyr/PlatformManagerImpl.cpp

[Telink] fixed B9x commissioning

Reverted network buffers configs

[Telink] TL3218 commissioning fixes

Changes to fix TLx commissioning on new Zephyr

[Telink] Zephyr revision update

After merge updates.
Kconfig change to fix TL3218 commissioning
Reduced RAM usage to build TL3218 by
COMMON_LIBC_MALLOC_ARENA_SIZE changing

[Telink] zephyr version change, cleanup

Changed MCUBOOT logs config, same for LZMA

interfer committed:
Merge pull request #349 from s07641069/update_zephyr_3.7_mem_optimize

[Telink] ROM usage optimization

interfer committed:
[Telink] W91 OTA, Factory data commissioning fix

Fixes full config target commissioning for W91.
Been failing on `SendAttestationRequest` step,
finished with `general error 0x01`.

Tested manually with chip-tool on RPi on tlsr9118bdk40d

interfer committed:
[Telink] W91 stacks increase

Increased NET_MGMT_EVENT_STACK_SIZE by 122 B
and IPC_SERVICE_BACKEND_ICMSG_WQ_STACK_SIZE by 256 B
so that net_mgmt and icmsg_workq stack were used <90%.
Jenkins reported thread analyzer warnings on that.

s07641069 commented on .github/workflows/examples-telink.yaml:
```diff
@@ -167,27 +178,27 @@ jobs:
+            - name: Build example Telink (B92) Light Switch App with OTA (LZMA), Shell, Factory Data, 4Mb flash
               # Run test for master and all PRs
               run: |
                   ./scripts/run_in_build_env.sh \
-                    "./scripts/build/build_examples.py --target 'telink-tlsr9528a-light-switch-ota-compress-lzma-shell-factory-data' build"
+                    "./scripts/build/build_examples.py --target 'telink-tlsr9528a-light-switch-ota-compress-lzma-shell-factory-data-4mb' build"
                   .environment/pigweed-venv/bin/python3 scripts/tools/memory/gh_sizes.py \
-                    telink tlsr9528a light-switch-app-ota-compress-lzma-shell-factory-data \
-                    out/telink-tlsr9528a-light-switch-ota-compress-lzma-shell-factory-data/zephyr/zephyr.elf \
+                    telink tlsr9528a light-switch-app-ota-compress-lzma-shell-factory-data-4mb \
+                    out/telink-tlsr9528a-light-switch-ota-compress-lzma-shell-factory-data-4mb/zephyr/zephyr.elf \
```
ok lets leave 4mb

interfer committed:
[Telink] review changes draft

Reverted NET_BUF_RX/TX_COUNT

s07641069 committed:
Merge branch 'master' into update_zephyr_3.7

andriy-bilynskyy committed:
[Telink] Add tl7218 board.

andriy-bilynskyy committed:
[Telink] Fix compilation warnings.

andriy-bilynskyy committed:
[Telink] Fix config option.

andriy-bilynskyy committed:
[Telink] Cleanup MbedTLS configuration.

s07641069 committed:
[Telink] Update tl721x default Kconfig

- update the default values for BT_RX_STACK_SIZE and COMMON_LIBC_MALLOC_ARENA_SIZE for TL721X

Signed-off-by: Jinmiao Yu <jinmiao.yu@telink-semi.com>

s07641069 committed:
[Telink] Remove unused files

s07641069 committed:
[Telink] Update Zephyr revision

s07641069 committed:
[Telink] Clean config types

s07641069 committed:
[Telink] Update Zephyr revision

s07641069 committed:
[Telink] Increase BLE_CTRL_THREAD_STACK_SIZE (B92 SoC retention test)

s07641069 committed:
[Telink] Fix compilation warninig

s07641069 committed:
[Telink] TL7218 upgrade to support version A2

s07641069 committed:
Merge branch 'master' into update_zephyr_3.7

s07641069 committed:
[Telink] Remove unused file after resolve merge conflict

s07641069 committed:
[Telink] Add tl7218_retention target

haiwentelink committed:
riscv: telink: seperate ram for tl3218x .

- use retention mode to seperate ram.
- reserve more ram for ilm.

Signed-off-by: Haiwen Xia <haiwen.xia@telink-semi.com>

serhiiSalamakha committed:
[Telink] adopt WiFi driver

s07641069 committed:
[Telink] Fix broken GitHub CI after adding tl721x target

s07641069 committed:
[Telink] remove temporary LZMA for tl321x due to bug

s07641069 committed:
[Telink] Fix TELINK_TLX_MATTER_RETENTION_LAYOUT typo for BOARD_TL3218

s07641069 committed:
[Telink] Add GPIO/PWM required for TL721X
https://github.com/telink-semi/hal_telink/commit/62edeab7cde786a1403cb062e7cef7d13edae520

s07641069 committed:
[Telink] Temporary disable B95 OTA failed build
B95 will be removed fully in next commits

s07641069 committed:
[Telink] Add missed clean out build output step

s07641069 committed:
[Telink] remove b95 SoC / TLSR9258a board (replaced by TL721x)

serhiiSalamakha committed:
[Telink] Revert InetUtils.cpp Telink dependency

s07641069 committed:
Merge branch 'master' into update_zephyr_3.7

s07641069 commented:
To minimize diff between [update_zephyr_3.7](https://github.com/s07641069/connectedhomeip/tree/update_zephyr_3.7) and [master](https://github.com/project-chip/connectedhomeip/tree/master) branches created next related PRs:
- https://github.com/s07641069/connectedhomeip/pull/382
- https://github.com/project-chip/connectedhomeip/pull/37356
- https://github.com/project-chip/connectedhomeip/pull/37357

s07641069 committed:
[Telink] zephyr versions compatibility changes (#382)

* [Telink] zephyr versions compatibility changes

* [Telink] Change Window Covering App default SoC
W91 doesn't support PM

* [Telink] Update Zephyr revision

* [Telink] sync B92 flash layout

* [Telink] Remove temp b92 4mb change

* [Telink] Avoid main stack owerflow for w91

* [Telink] Update Zephyr revision

s07641069 committed:
Merge branch 'master' into update_zephyr_3.7

s07641069 committed:
Merge branch 'master' into update_zephyr_3.7

s07641069 committed:
[Telink] manual sync after merge master

s07641069 committed:
[Telink] Increase Main Stack size for W91
Update Zephyr revision

s07641069 committed:
[Telink] Zephyr 3.3 backward compatibility support

fengtai-telink committed:
riscv: telink: fix insufficient RAM when config LZMA (#384)

 - adjust the zephyr hash to 930b6d4 .

Signed-off-by: Fengtai Xie <fengtai.xie@telink-semi.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=36691) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=36691) before we can accept your contribution.<br/>**5** out of **6** committers have signed the CLA.<br/><br/>:white_check_mark: andriy-bilynskyy<br/>:white_check_mark: s07641069<br/>:white_check_mark: interfer<br/>:white_check_mark: serhiiSalamakha<br/>:white_check_mark: haiwentelink<br/>:x: fengtai-telink<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/project-chip/connectedhomeip?pullRequest=36691) it.</sub>

fengtai-telink committed:
riscv: telink: add retention mode for buteo (#385)

* riscv: telink: Update Zephyr hash

 - adjust the zephyr hash to 9e39d9f .
 - add tl3218x_retention target .

Signed-off-by: Fengtai Xie <fengtai.xie@telink-semi.com>

* riscv: telink: remove retention build target for buteo

Signed-off-by: Fengtai Xie <fengtai.xie@telink-semi.com>

* [Telink] Add comment

---------

Signed-off-by: Fengtai Xie <fengtai.xie@telink-semi.com>
Co-authored-by: Alex Tsitsiura <s07641069@gmail.com>

s07641069 committed:
[Telink] Fix commisioning issue due to missed MbedTLS module
TODO: fix compilation warnings

s07641069 committed:
telink: tl7218x: fix tercel retention (Update Zephyr hash)

s07641069 committed:
[Telink] Optimize add MbedTLS definitions for Zephyr 3.3 without warnings

s07641069 committed:
Merge branch 'master' into update_zephyr_3.7

s07641069 committed:
Merge branch 'master' into update_zephyr_3.7

s07641069 committed:
[Telink]  Add zephyr_3.3 job

s07641069 committed:
[Telink] Fix job name

s07641069 committed:
[Telink] test same job name to avoid settings changes

s07641069 committed:
[Telink] Use Shell for tl321x lighting build (Zephyr 3.3)

s07641069 committed:
[Telink] Decrease number of CI builds for all PRs

s07641069 committed:
[Telink] Update readme files

s07641069 committed:
Merge branch 'master' into update_zephyr_3.7

s07641069 committed:
Update builds to docker version 121

restyled-commits committed:
Restyled by prettier-markdown

s07641069 committed:
Revert TI CC32XX image

s07641069 commented:
@andy31415 I tried to decrease as much as possible numbers of CI build for all PRs (allowed for PRs from my fork).
See description, there I provided table for easy check which (how mutch) build are enabled for "all PRs".

s07641069 committed:
Update builds to docker version 122

s07641069 committed:
[Telink] Add Thread Optimization Configs

- OPENTHREAD_CSMABACKOFF_OPTIMIZATION
- IEEE802154_TLX_OPTIMIZATION
- set COMMON_LIBC_MALLOC_ARENA_SIZE to
20kb if tl7218x_retention for power consumption optimization
based on d6882ec0719692a1ef1bc983e138f32bf1a39442.

s07641069 committed:
[Telink] Fix duplicated name of Size Reports

s07641069 committed:
Merge branch 'master' into update_zephyr_3.7

s07641069 committed:
Merge branch 'master' into update_zephyr_3.7

s07641069 committed:
Update builds to docker version 123

s07641069 committed:
keep Tizen on 122

s07641069 committed:
Merge branch 'master' into update_zephyr_3.7

s07641069 committed:
Update builds to docker version 125

s07641069 committed:
revert cc32xx

s07641069 commented:
@andy31415 could you review this PR?
